packages/ui/src/common/components/forms/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,182 @@
+import { addMonths, addWeeks, addYears, isAfter, isBefore, isEqual, startOfMonth, startOfToday } from 'date-fns'
+import React, { useEffect, useMemo, useRef, useState } from 'react'
+import styled from 'styled-components'
+
+import { ButtonSecondary, FilterButtons } from '@/common/components/buttons'
+import { Shadows } from '@/common/constants'
+import { DateRange, PartialDateRange } from '@/common/types/Dates'
+import { earliest, fromRange, latest, toDDMMYY } from '@/common/utils/dates'
+
+import { Calendar } from '../Calendar'
+import { InputComponent, InputText } from '../InputComponent'
+
+interface DatePickerProps {
+  title?: string
+  value: PartialDateRange
+  withinDates?: PartialDateRange
+  onApply?: () => void
+  onClear?: () => void
+  onChange: (value: PartialDateRange) => void
+}
+
+export const DatePicker = ({ title, value, withinDates, onApply, onClear, onChange }: DatePickerProps) => {
+  const placeholder = '__/__/__'
+  const { start, end } = fromRange(value)
+  const dateString = `${toDDMMYY(start) ?? placeholder} - ${toDDMMYY(end) ?? placeholder}`
+
+  const container = useRef<HTMLDivElement>(null)
+  const [isOpen, toggleOpen] = useState(false)
+
+  useEffect(() => {
+    if (!isOpen) return
+
+    const closePopup = (evt: MouseEvent) => {
+      evt.stopPropagation()
+      if (!evt.target) return
+
+      const target = evt.target as Node
+      const clickedOutside = !container.current?.contains(target) ?? false
+      clickedOutside && toggleOpen(false)
+    }
+
+    window.addEventListener('mousedown', closePopup)
+    return () => window.removeEventListener('mousedown', closePopup)
+  }, [isOpen])
+
+  const apply = () => {
+    onApply?.()
+    toggleOpen(false)
+  }
+
+  return (
+    <DatePickerContainer ref={container} onMouseDown={() => !isOpen && toggleOpen(true)}>
+      <InputComponent tight inputWidth="xs">
+        <InputText placeholder="-" value={dateString} readOnly />
+        {isOpen && (
+          <DatePickerPopup>
+            <DatePickerCalendars value={value} within={withinDates} onChange={onChange} />
+            <DatePickerFooter>
+              <FilterButtons onApply={apply} onClear={onClear} />
+            </DatePickerFooter>
+          </DatePickerPopup>
+        )}
+      </InputComponent>
+    </DatePickerContainer>
+  )
+}
+
+interface DatePickerCalendarsProps {
+  value: PartialDateRange
+  within?: PartialDateRange
+  onChange: (value: PartialDateRange) => void
+}
+const DatePickerCalendars = ({ value, within, onChange }: DatePickerCalendarsProps) => {
+  const { start, end } = fromRange(value)
+
+  const today = useMemo(() => startOfToday(), [])
+  const thisMonth = useMemo(() => startOfMonth(today), [])
+  const [monthL, setMonthL] = useState(start ? startOfMonth(start) : addMonths(thisMonth, -1))
+  const [monthR, setMonthR] = useState(end ? startOfMonth(end) : thisMonth)
```

Comment:
If you could rename `mothL` & `monthR` -> the abbreviation is a bit cryptic. It will be better to be a bit more fluent with variable names, even if longer. I'd guess that is `monthLeft` and `monthRight` but if this has better meaning, like `startMonth` & `endMonth` the better. (Because Left & RIght is a visual aspect of those two variables, but their left/right placement has different meaning.

Reply:
My understanding from the design is that both calendars can be changed (to some extent) independently from each other and from the date range. But also you could use e.g the left calendar to set both the start and end dates so endMonth could be a bit confusing. In any case I think `leftCalendarMonth` and `rightCalendarMonth` are much clearer names than the previous ones :smile: